### PR TITLE
WT-11888 Fix Evergreen Python version detection on macOS

### PR DESCRIPTION
**Pull request by korteland**
CMake obsoleted the non-versioned Python executable/library/include strings, plus the code for explicitly setting those wasn't running for the arm64 variant.

### CODE REVIEW

**Comment by korteland** [comment]
Added all macOS tests to PR patch build.

File: test/evergreen.yml [hunk 1]
```diff
@@ -195,7 +195,7 @@ functions:
           DEFINED_EVERGREEN_CONFIG_FLAGS=${DEFINED_EVERGREEN_CONFIG_FLAGS/\-DHAVE_BUILTIN_EXTENSION_ZSTD=1/}
         fi
 
-        if [ "${build_variant|}" = "macos-1100" ]; then
+        if [ "${build_variant|}" = "macos-1100" -o "${build_variant|}" = "macos-1100-arm64" ]; then
```
Comment (jiechenbo) [inline test/evergreen.yml:198]: what do you think about converting into a regex and look for only macos-1100 here?

Comment (korteland) [reply test/evergreen.yml]: Thanks, good call - done in 933838f.

**Review by jiechenbo — APPROVED** [review]
LGTM!

**Comment by korteland** [comment]
I've pushed a fix for the x86 versions of macOS - that was trickier than I thought since they're running a super old CMake that does lots of stuff differently. So the build scripts diverged more than I thought they needed to. Anyway - my patch build is now working for both macOS architectures.

File: test/evergreen.yml [hunk 2]
```diff
@@ -214,13 +214,18 @@ functions:
...
+               SYSPYINCDEF="$SYSPYDIR/Headers"
              fi
           fi
-          DEFINED_EVERGREEN_CONFIG_FLAGS="$DEFINED_EVERGREEN_CONFIG_FLAGS -DPYTHON_EXECUTABLE:FILEPATH=$SYSPY -DPYTHON_LIBRARY=$SYSPYLIB $SYSPYINCDEF"
+
+          if [ "${build_variant|}" = "macos-1100-arm64" ]; then
+            DEFINED_EVERGREEN_CONFIG_FLAGS="$DEFINED_EVERGREEN_CONFIG_FLAGS -DPython3_EXECUTABLE=$SYSPY -DPython3_LIBRARY=$SYSPYLIB -DPython3_INCLUDE_DIR=$SYSPYINCDEF"
+          else
+            DEFINED_EVERGREEN_CONFIG_FLAGS="$DEFINED_EVERGREEN_CONFIG_FLAGS -DPYTHON_EXECUTABLE:FILEPATH=$SYSPY -DPYTHON_LIBRARY=$SYSPYLIB -DPYTHON_INCLUDE_DIR=$SYSPYINCDEF"
+          fi
```
Comment (jiechenbo) [inline test/evergreen.yml:228]: Nice find! I was thinking we could create a variable specifically for the `-DPython3_EXECUTABLE=$SYSPY` configuration. For example :

```
          if [ "${build_variant|}" = "macos-1100-arm64" ]; then
            PYTHON_EXEC_FLAG="-DPython3_EXECUTABLE=$SYSPY"
          else
            PYTHON_EXEC_FLAG="-DPython3_EXECUTABLE:FILEPATH=$SYSPY"
          fi
          
          DEFINED_EVERGREEN_CONFIG_FLAGS="$DEFINED_EVERGREEN_CONFIG_FLAGS $PYTHON_EXEC_FLAG -DPYTHON_LIBRARY=$SYSPYLIB -DPYTHON_INCLUDE_DIR=$SYSPYINCDEF"
```

Let me know what you think.

Comment (korteland) [reply test/evergreen.yml:228]: I tried that at one point. Unfortunately there are other differences - `PYTHON_LIBRARY` and `PYTHON_INCLUDE_DIR` need to be spelled differently across the versions.

**Review by jeremythorp — APPROVED** [review]
LGTM!

Hopefully one day the Intel and ARM Macs will have the same configs and we can simplify this.

**Comment by korteland** [comment]
evergreen merge

**Pull request merged**